platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/recipe/RecipeLoader.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.machine.server.recipe;
+
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.machine.server.dao.RecipeDao;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+/**
+ * Loads predefined recipes into local dao.
+ *
+ * @author Anton Korneta
+ */
+@Singleton
+public class RecipeLoader {
+    private final String    recipesName;
+    private final RecipeDao recipeDao;
+    private final Gson      gson;
+
+    @Inject
+    public RecipeLoader(@Named("codenvy.local.infrastructure.recipes") String recipesName, RecipeDao recipeDao) {
+        this.recipesName = recipesName;
+        this.recipeDao = recipeDao;
+        this.gson = new Gson();
+    }
+
+    @PostConstruct
+    public void start() throws ServerException {
+        URL recipeUrl = Thread.currentThread().getContextClassLoader().getResource(recipesName);
+        if (recipeUrl == null) {
+            throw new ServerException("Impossible get predefined recipes");
```

Comment:
I think this message should be more informative 
i.e. _File with predefined recipes '/path/to/recipes.json' doesn't exist_